src/Stream.php

```diff
@@ -243,7 +243,7 @@ public function appendFilter(string $filtername, int $read_write, $params = null
             return;
         }
 
-        throw new Exception(sprintf('unable to locate filter `%s`', $filtername));
+        throw new CannotAddStreamFilter(sprintf('unable to locate filter `%s`', $filtername));
```

Comment:
This should be a `InvalidArgument` exception. Stream Filtering is supported *but* the filter name is invalid.